### ISSUE TITLE
A dataclass of type ``enum.Enum`` does not contain the enum ``__members__`` object

### ISSUE THREAD

**Issue by mbyrnepr2** [pylint-dev/astroid#2317]
### Steps to reproduce
See https://github.com/pylint-dev/pylint/issues/9100 for context/original report.
A fix is made to Pylint to prevent the crash but work required in astroid to fix the root cause.

1.
2.
3.

### Current behavior
A false negative in Pylint for ``invalid-name``:
```python
from dataclasses import dataclass
from enum import Enum


@dataclass
class Something(str, Enum):
	asd: str
```

### Expected behavior
No false negative

### `python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"` output
3.1.0-dev0

**Comment by jacobtylerwalls** [comment]
Possibly related: #1730

**Comment by mbyrnepr2** [comment]
I don't think so @jacobtylerwalls. That issue is basically about excluding dunders from `__members__` (it isn't necessarily about data classes) while this issue is specifically about `__members__` not being available at all when dataclass decorator is used.

**Comment by DanielNoord** [comment]
Pitching https://github.com/pylint-dev/astroid/pull/1598 again as it would be nice to broaden our support for instance variables 😄 


Note that this is an issue in `astroid`, `__members__` does exist:
```
from dataclasses import dataclass
from enum import Enum


@dataclass
class Something(str, Enum):
	asd: str = 1  # [invalid-name]

class OtherSomething(str, Enum):
    asd: str = 1  # [invalid-name]

print(Something.__members__)
print(OtherSomething.__members__)
```

```console
❯ python test.py
{'asd': <Something.asd: '1'>}
{'asd': <OtherSomething.asd: '1'>}
```

**Comment by mbyrnepr2** [comment]
Indeed @DanielNoord. Perhaps I was unclear in the comment. I meant that `__members__` isn't present on the _astroid_ representation of the class. I agree that it exists on the Python class.
